Update functions and install for flatpak steam installation

bradleygit opened the pull request:
Came across this issue when trying to install 7th Heaven on my Fedora machine with Flatpak steam only.

Running the script with the Flatpak version of steam causes it not to install and fail. I've updated the getSteamLibrary function and install script to accomodate. I have tested flatpak and non flatpak 7th heaven installs and they work as intended. Flatpak has some caveats, I could not get it to automatically as a non-steam game and to get it to automatically start FF7 to generate the prefix (and then kill it). The user can add the game via GUI method in steam.

Image is running via 7th heaven with flatpak only steam
![Flatpak](https://github.com/user-attachments/assets/592380dc-5f1c-43ed-a2a3-df411bd15017)

dotaxis commented:
There are a couple of reasons I've not added flatpak support to 7thDeck so far: 

1. I wasn't sure if there would be issues with 7th Heaven actually working if flatpak Steam was used.
2. I wasn't able to easily make the script behave the same way for both flatpak and native installations of Steam
3. I am not sure I want to have to deal with supporting flatpak Steam in the long term in case something breaks, since I have no intention of using flatpak Steam on any of my Linux devices


All of that said, I like what you've done here and will have a closer look tomorrow. If you can solve the problem of automatically launching FF7 (try `flatpak run com.valvesoftware.Steam steam://rungameid/39140` to start) and also add a prompt to handle cases where both flatpak and native Steam are installed on the same machine I think we can move forward with this. 

I will try to solve the "add to Steam" problem but honestly it only works half the time for native Steam so it's not a big deal if we can't get it working.

bradleygit commented:
Thanks for looking at my PR 😃 the change was mostly just to get it installed on my machine and thought to PR it if anyone else has the same issue. I can definitely fix the case with normal and flatpak install later today and I believe I tried the flatpak run command but it just brought the steam GUI to the front without launching but I will have another look at this also

dotaxis commented on install.sh:
```diff
@@ -100,16 +111,29 @@ while pidof "steam" > /dev/null; do
-perl -0777 -i -pe 's/"CompatToolMapping"\n\s+{/"CompatToolMapping"\n\t\t\t\t{\n\t\t\t\t\t"39140"\n\t\t\t\t\t{\n\t\t\t\t\t\t"name"\t\t"proton_9"\n\t\t\t\t\t\t"config"\t\t""\n\t\t\t\t\t\t"priority"\t\t"250"\n\t\t\t\t\t}/gs' \
-$HOME/.steam/steam/config/config.vdf
+
+if [ $IS_FLATPAK = true ]; then
```
This whole block is redundant since `STEAMROOT` is determined above.

We only need to change line 105 to use `$STEAMROOT` instead of `$HOME/.steam/steam`

dotaxis commented on install.sh:
```diff
@@ -3,7 +3,16 @@
 XDG_DATA_HOME="${XDG_DATA_HOME:=${HOME}/.local/share}"
 IS_STEAMOS=$(grep -qi "SteamOS" /etc/os-release && echo true || echo false)
+IS_FLATPAK=false
+STEAMROOT=$(echo $HOME/.steam/steam)
```
This should be `STEAMROOT=$(readlink -f "$HOME/.steam/root")`

dotaxis commented on install.sh:
```diff
@@ -3,7 +3,16 @@
+else
+  echo "Steam is installed via Flatpak."
+  IS_FLATPAK=true
+  STEAMROOT=$(echo $HOME/.var/app/com.valvesoftware.Steam/.local/share/Steam)
```
The echo here is redundant, you can just use
`STEAMROOT="$HOME/.var/app/com.valvesoftware.Steam/.local/share/Steam"`

dotaxis reviewed (changes requested):
There are a couple of things here that should be changed and I've left comments to address those specifically.

Also, not to nitpick, but can you undo the changes to the indentation in `functions.sh` and also remove the extra echo statements and blank lines you added in `install.sh`? Namely lines 38, 40. 160-161 and 186.

Once you make these changes I can merge and I will figure out a way to handle cases where people have flatpak and native steam installed on the same machine (because you can always count on people doing weird stuff like that).

bradleygit commented:
Completed the changes from above, fixed one additional issue with flatpak steam not loading correctly after being killed and reloaded. Added the choice if both are found on one system (screenshot for this added) 
![bothInstalled](https://github.com/user-attachments/assets/87e120ab-df4d-440f-81c8-0316d1e20941)

dotaxis commented on install.sh:
```diff
@@ -3,6 +3,39 @@
+IS_FLATPAK=false
+STEAMROOT=$(readlink -f "$HOME/.steam/root")
+
+if ! $(flatpak list --app | grep -q Steam && echo true || echo false); then
+  echo "Steam is not installed via Flatpak."
+else
+  echo "Steam is installed via Flatpak."
```
Can you send these two echos to the log file instead of printing them to the console? We don't need to display this info on every install. Just add `&> "7thDeck.log"` to the end of the lines.

dotaxis reviewed (changes requested):
Looking good! Please move the block from line 9-38 so it appears under the banner, and remove the blank lines at 136 and 140.

After these changes I will test and merge tomorrow. :)

Thank you!

bradleygit commented:
Completed those changes :)

dotaxis commented:
Awesome. I'll test it out in the morning and merge in the afternoon. :)

Thank you!
